oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticIndexTracker.java

```diff
@@ -44,4 +52,19 @@ public boolean isUpdateNeeded(NodeState before, NodeState after) {
     protected ElasticIndexNodeManager openIndex(String path, NodeState root, NodeState node) {
         return new ElasticIndexNodeManager(elasticConnection, path, root);
     }
+
+    @Override
+    protected Set<String> getIndexPaths(final NodeState root) {
+        Set<String> indexPaths = super.getIndexPaths(root);
```

Comment:
@fabriziofortino I wanted to confirm if this is what we discussed earlier - checking if the update is for an elastic index definition and add it to index tracker. 
This seems to work but this is called for every commit and this code will traverse all the index definitions under oak:index. Doesn't seem very efficient.

If we need to traverse, then it would be better if we just do that when clean up runs (once every 5s by default).

Reply:
@averma21 yes, this is pretty much what I proposed.
Apart from that, I was thinking if we can simplify the overall design and logic. Not sure if this will be good in all cases, but I was thinking the following:
* remove the logic introduce in `AsyncIndexUpdate`
* pass a `onDeleteIndexDefinition` callback to `ElasticIndexTracker` that removes the index from Elastic
* the `ElasticIndexTracker` can call the callback at initialization time (if detects an existing index in Elastic has no index definition associated) or when an index definition gets removed.

This will remove the need to check every 5 secs if we need to remove indexes.

Reply:
The problem with observer logic is that deletion event would be triggered only once but if elastic server is unavailable at that time, we would not be able to delete that index unless we add some kind of retry mechanism. That is why this sort of "garbage collection" is being used.

Reply:
That's true. This is why I proposed to do the check also when IndexTracker initializes (it would be called every time a cluster node gets started). I was just trying to see if there could be ways to (a) avoid making a call to Elastic every 5s (b) implement a synchronous delete (this will be handy when we need to delete indexes after performance tests since in async mode we should delete the definition and wait for at least 5s).